src/main/java/doke/TaskList.java

```diff
@@ -0,0 +1,173 @@
+package doke;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import java.time.format.DateTimeFormatter;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * A class to represent a TaskList.
+ */
+public class TaskList {
+
+    private ArrayList<Task> taskList = new ArrayList<>();
+
+    /**
+     * A public constructor for the TaskList class.
+     * It will try to read and create and ArrayList of type task from storage,
+     * which, if not possible would then create an empty ArrayList of type Task.
+     *
+     * @param ui
+     * @param storage
+     */
+    public TaskList(Ui ui, Storage storage) {
```

Comment:
This constructor is slightly long and the control flow may be hard to follow due to some inverted statements.
Perhaps consider shifting the Scanner responsibilities to a new method/class?